build-locally.sh

```diff
@@ -0,0 +1,732 @@
+    --releaseMetadata ${WORKSPACE_DIR}/galasa/modules/extensions/release.yaml \
+    --releaseMetadata ${WORKSPACE_DIR}/galasa/modules/managers/release.yaml \
+    --releaseMetadata ${WORKSPACE_DIR}/galasa/modules/obr/release.yaml \
+    --template pom.template \
+    --output pom.xml \
+    --isolated"
+
+    echo "Command is $cmd" >> ${log_file}
+    $cmd 2>&1 >> ${log_file}
+
+    rc=$?
+    if [[ "${rc}" != "0" ]]; then
+        error "Failed to generate a pom.xml from the pom.template. Log file is ${log_file}"
+        exit 1
+    fi
+    success "pom.xml generated ok - log is at ${log_file}"
+}
+
+#------------------------------------------------------------------------------------
+function build_pom_xml {
+
+    export DIRECTORY=$1
+    export DISTRIBUTION=$2
+
+    cd ${DIRECTORY}
+
+    h2 "Building the pom.xml for ${DISTRIBUTION}"
+
+    cmd="mvn -f pom.xml process-sources -X \
+    -Dgpg.skip=true \
+    -Dgalasa.target.repo=file:target/isolated/maven \
+    -Dgalasa.runtime.repo=${SOURCE_MAVEN_OBR} \
+    -Dgalasa.simplatform.repo=${SOURCE_MAVEN_SIMPLATFORM} \
+    -Dgalasa.javadoc.repo=${SOURCE_MAVEN_JAVADOC} \
+    -Dgalasa.docs.repo=https://maven.pkg.github.com/galasa-dev/galasa.dev \
+    -Dgalasa.central.repo=https://repo.maven.apache.org/maven2/ \
+    -Dgithub.token.read.packages.username=${GITHUB_USERNAME} \
+    -Dgithub.token.read.packages.password=${GITHUB_TOKEN} \
+    --batch-mode --errors --fail-at-end \
+    --settings ${BASEDIR}/settings.xml"
+
+    echo "Command is $cmd" >> ${log_file}
+    $cmd 2>&1 >> ${log_file}
+
+    rc=$?
+    if [[ "${rc}" != "0" ]]; then
+        error "Failed to build pom.xml. Log file is ${log_file}"
+        exit 1
+    fi
+    success "pom.xml built ok - log is at ${log_file}"
+
+}
+
+#------------------------------------------------------------------------------------
+function build_pom2_xml {
+
+    export DIRECTORY=$1
+    export DISTRIBUTION=$2
+
+    cd ${DIRECTORY}
+
+    h2 "Building the pom2.xml for ${DISTRIBUTION}"
+
+    cmd="mvn -f pom2.xml process-sources -X \
+    -Dgpg.skip=true \
+    -Dgalasa.target.repo=file:target/isolated/maven \
+    -Dgalasa.runtime.repo=${SOURCE_MAVEN_OBR} \
+    -Dgalasa.simplatform.repo=${SOURCE_MAVEN_SIMPLATFORM} \
+    -Dgalasa.javadoc.repo=${SOURCE_MAVEN_JAVADOC} \
+    -Dgalasa.docs.repo=https://maven.pkg.github.com/galasa-dev/galasa.dev \
+    -Dgalasa.central.repo=https://repo.maven.apache.org/maven2/ \
+    -Dgithub.token.read.packages.username=${GITHUB_USERNAME} \
+    -Dgithub.token.read.packages.password=${GITHUB_TOKEN} \
+    --batch-mode --errors --fail-at-end \
+    --settings ${BASEDIR}/settings.xml"
+
+    echo "Command is $cmd" >> ${log_file}
+    $cmd 2>&1 >> ${log_file}
+
+    rc=$?
+    if [[ "${rc}" != "0" ]]; then
+        error "Failed to build pom2.xml. Log file is ${log_file}"
+        exit 1
+    fi
+    success "pom2.xml built ok - log is at ${log_file}"
+
+}
```

Comment:
These `build_pomx_xml` functions look really similar, could we perhaps make the `pom.xml` file a parameter so that we could just use one build function to build multiple poms? (happy to keep as is if there's a reason we've split them up)

Reply:
Thank you great point - I've now made one function to build all the poms that do the `process-sources` goal and pass in the parameters.